### DISCUSSION TITLE
Usage query

### DISCUSSION THREAD

**Discussion by ursanjeev** [grafana/grafana-infinity-datasource#175] [Q&A]
Hi, the rest api I am using provides the below output:

`{
  "return": {
    "message": "Success",
    "code": 200,
    "data": "[{\"timestamp\": \"2021-10-04 08:42:09\", \"uptime\": 27485260.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 08:47:09\", \"uptime\": 27488260.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 08:52:09\", \"uptime\": 27491260.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 08:57:09\", \"uptime\": 27494262.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 09:02:09\", \"uptime\": 27497271.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 09:07:09\", \"uptime\": 27500262.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 09:12:09\", \"uptime\": 27503274.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 09:17:09\", \"uptime\": 27506271.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 09:22:09\", \"uptime\": 27509263.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 09:27:09\", \"uptime\": 27512261.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 09:32:09\", \"uptime\": 27515262.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 09:37:09\", \"uptime\": 27518281.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 09:42:09\", \"uptime\": 27521262.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 09:47:09\", \"uptime\": 27524292.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 09:52:09\", \"uptime\": 27527262.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 09:57:09\", \"uptime\": 27530262.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 10:02:09\", \"uptime\": 27533263.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 10:07:09\", \"uptime\": 27536273.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 10:12:09\", \"uptime\": 27539263.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 10:17:09\", \"uptime\": 27542264.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 10:22:09\", \"uptime\": 27545263.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 10:27:09\", \"uptime\": 27548263.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 10:32:09\", \"uptime\": 27551262.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 10:37:09\", \"uptime\": 27554265.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 10:42:09\", \"uptime\": 27557267.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 10:47:09\", \"uptime\": 27560323.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 10:52:09\", \"uptime\": 27563266.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 10:57:09\", \"uptime\": 27566264.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 11:02:09\", \"uptime\": 27569263.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 11:07:09\", \"uptime\": 27572263.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 11:12:09\", \"uptime\": 27575277.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 11:17:09\", \"uptime\": 27578263.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 11:22:09\", \"uptime\": 27581268.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 11:27:09\", \"uptime\": 27584264.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 11:32:09\", \"uptime\": 27587264.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 11:37:09\", \"uptime\": 27590267.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 11:42:09\", \"uptime\": 27593268.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 11:47:09\", \"uptime\": 27596264.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 11:52:09\", \"uptime\": 27599267.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 11:57:09\", \"uptime\": 27602275.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 12:02:09\", \"uptime\": 27605267.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 12:07:09\", \"uptime\": 27608277.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 12:12:09\", \"uptime\": 27611272.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 12:17:09\", \"uptime\": 27614267.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 12:22:09\", \"uptime\": 27617266.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 12:27:09\", \"uptime\": 27620267.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 12:32:09\", \"uptime\": 27623265.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 12:33:59\", \"uptime\": 27624381.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 12:37:09\", \"uptime\": 27626268.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 12:47:09\", \"uptime\": null, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 12:57:09\", \"uptime\": null, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 13:07:09\", \"uptime\": null, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 13:17:09\", \"uptime\": null, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 13:27:09\", \"uptime\": null, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 13:37:09\", \"uptime\": null, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 13:47:09\", \"uptime\": null, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 13:52:09\", \"uptime\": 27671271.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 13:57:10\", \"uptime\": 27674277.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 14:02:09\", \"uptime\": 27677268.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 14:07:09\", \"uptime\": 27680281.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 14:12:09\", \"uptime\": 27683267.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 14:17:10\", \"uptime\": 27686290.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 14:22:09\", \"uptime\": 27689280.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 14:27:09\", \"uptime\": 27692274.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 14:32:09\", \"uptime\": 27695269.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 14:37:09\", \"uptime\": 27698268.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 14:42:09\", \"uptime\": 27701270.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 14:47:09\", \"uptime\": 27704270.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 14:52:09\", \"uptime\": 27707272.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 14:57:09\", \"uptime\": 27710272.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 15:02:09\", \"uptime\": 27713280.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 15:07:09\", \"uptime\": 27716277.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 15:12:09\", \"uptime\": 27719283.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 15:17:09\", \"uptime\": 27722270.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 15:22:09\", \"uptime\": 27725270.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 15:27:09\", \"uptime\": 27728270.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 15:32:09\", \"uptime\": 27731272.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 15:37:09\", \"uptime\": 27734272.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 15:42:09\", \"uptime\": 27737271.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 15:47:09\", \"uptime\": 27740271.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 15:52:09\", \"uptime\": 27743285.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 15:57:09\", \"uptime\": 27746271.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 16:02:09\", \"uptime\": 27749273.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 16:07:09\", \"uptime\": 27752273.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 16:12:10\", \"uptime\": 27755280.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 16:17:09\", \"uptime\": 27758273.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 16:22:09\", \"uptime\": 27761272.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 16:27:09\", \"uptime\": 27764271.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 16:32:09\", \"uptime\": 27767274.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 16:37:09\", \"uptime\": 27770272.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 16:42:09\", \"uptime\": 27773271.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 16:47:09\", \"uptime\": 27776273.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 16:52:09\", \"uptime\": 27779274.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 16:57:09\", \"uptime\": 27782272.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 17:02:09\", \"uptime\": 27785275.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 17:07:09\", \"uptime\": 27788273.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 17:12:09\", \"uptime\": 27791275.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 17:17:09\", \"uptime\": 27794275.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}, {\"timestamp\": \"2021-10-04 17:22:09\", \"uptime\": 27797274.0, \"period\": null, \"filename\": \"cpe.testCPE4.snmp\"}]"
  }
}`

I tried configuring the Infinity Data source to consume this and allow me to select the columns such as timestamp, uptime etc..., but I am not able to. Can you please advise how to use the Infinity data source to consume this result? 

Also how can I pass a parameter to the API via the body of the query? I have the query parameter stored in a Grafana dashboard variable.

Many thanks!

Sanjeev G

**Comment by yesoreyeram** [comment]
Your data field doesn't seems to be JSON. It seems to be a JSON wrapped as a string. So it is not possible unless your data is returned as actual json.

**Reply by ursanjeev** [reply]
Thanks! Is there a way to consume this data using the Infinity data source?

**Reply by yesoreyeram** [reply]
Yes. But only if the server send this as json field instead string.
